File: practice/algorithms/implementation/modified_kaprekar_numbers/ModifiedKaprekarNumbers.js
'use strict';

const processData = input => {
    const lines = input.split('\n').map(i => parseInt(i));
    const lower = lines[0];
    const upper = lines[1];
    const kaprekarNums = [];
    for(let i = lower; i <= upper; i++) {
        if(isKaprekar(i)) {
            kaprekarNums.push(i);
        }
    }
    if(kaprekarNums.length > 0) {
        console.log(kaprekarNums.join(" "));
    } else {
        console.log("INVALID RANGE");
    }
};

const isKaprekar = (num) => {
    const square = parseInt(Math.pow(num, 2)).toString();
    if(square.length > 1) {
        const l = parseInt(square.substring(0, square.length / 2));
        const r = parseInt(square.substring(square.length / 2));
        return (l + r) === num;
    } else {
        return parseInt(square) === num;
    }
}

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/modified_kaprekar_numbers/modified_kaprekar_numbers.rb
def is_kaprekar(num)
  square = (num ** 2).to_s
  if(square.size > 0)
    l = square.slice(0, (square.size / 2).to_i).to_i
    r = square.slice((square.size / 2).to_i, (square.size)).to_i
    return (l + r) == num
  else
    return square.to_i == num
  end
end

lower = gets.to_i
upper = gets.to_i
kaprekars = []
for i in lower..upper
  kaprekars << i if(is_kaprekar(i))
end

if(kaprekars.size > 0)
  puts kaprekars.join(" ")
else
  puts "INVALID RANGE"
end